server/src/main/java/com/linecorp/centraldogma/server/metadata/Roles.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.centraldogma.server.metadata;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+
+import com.linecorp.centraldogma.common.RepositoryRole;
+
+/**
+ * Role metadata for a repository.
+ */
+public final class Roles {
+
+    private final ProjectRoles projectRoles;
+
+    private final Map<String, RepositoryRole> users;
+
+    private final Map<String, RepositoryRole> tokens;
+
+    /**
+     * Creates a new instance.
+     */
+    @JsonCreator
+    public Roles(@JsonProperty("projects") ProjectRoles projectRoles,
+                 @JsonProperty("users") Map<String, RepositoryRole> users,
+                 @JsonProperty("tokens") Map<String, RepositoryRole> tokens) {
+        this.projectRoles = requireNonNull(projectRoles, "projectRoles");
+        this.users = requireNonNull(users, "users");
+        this.tokens = requireNonNull(tokens, "tokens");
+    }
+
+    /**
+     * Returns the {@link ProjectRoles} of the repository.
+     */
+    @JsonProperty("projects")
+    public ProjectRoles projectRoles() {
```

Comment:
I understood this property as: the  default `RepositoryRole` that will be assigned to project members that aren't explicitly set by `users` and `tokens`.

Reply:
> the default RepositoryRole that will be assigned to project members

That's correct. 👍 

> that aren't explicitly set by users and tokens.

That is not correct. A `RepositoryRole` can still be assigned to a user. The `RepositoryRole`, which operates at a higher level, will be applied in such situation. https://github.com/line/centraldogma/pull/1060/files#diff-dc8a031ec8b91f16d76b42e152e9c3be8dd07b6f77bea922030074acc78ce78eR800-R823